src/runtime/HalideRuntime.h

```diff
@@ -541,6 +546,38 @@ extern int halide_copy_to_host(void *user_context, struct halide_buffer_t *buf);
 extern int halide_copy_to_device(void *user_context, struct halide_buffer_t *buf,
                                  const struct halide_device_interface_t *device_interface);
 
+/** Copy data from one buffer to another. The buffers may have
+ * different shapes and sizes, but the destination buffer's shape must
+ * be contained within the source buffer's shape. The source data is
+ * pulled from either device or host memory on the source, depending
+ * on the dirty flags. host is preferred if both are valid. The
+ * dst_device_interface parameter controls the destination memory
+ * space. NULL means host memory. */
+extern int halide_buffer_copy(void *user_context, struct halide_buffer_t *src,
+                              const struct halide_device_interface_t *dst_device_interface,
+                              struct halide_buffer_t *dst);
+
+/** Give the destination buffer a device allocation which is an alias
+ * for the same coordinate range in the source buffer. Modifies the
+ * device, device_interface, and the device_dirty flag only. Only
+ * supported by some device APIs (others will return
+ * halide_error_code_device_crop_unsupported). Do not call device_free
+ * on the resulting buffer. Call halide_device_release_crop instead to
+ * clean up any resources associated with the cropped view. Do not
```

Comment:
Comment fixed. I want to keep things consistent with wrap/detach device_and_host_malloc/device_and_host_free and device_malloc/device_free for now, which use separate pairs. Buffer currently messes up these cases. It needs an enum to track what it should do with the device pointer on destruction.